src/com/amazon/ionelement/impl/BinaryIonElement.kt

```diff
@@ -16,11 +16,13 @@
 package com.amazon.ionelement.impl
 
 import com.amazon.ionelement.api.IonByteArray
+import com.amazon.ionelement.api.LobElement
 
-internal abstract class BinaryIonElement: IonElementBase(), IonByteArray {
-    protected abstract val bytes: ByteArray
+internal abstract class BinaryIonElement(
+    protected val bytes: ByteArray
+): AnyElementBase(), IonByteArray, LobElement {
 
-    override val bytesValueOrNull: IonByteArray get() = this
+    override val bytesValue: IonByteArray by lazy (LazyThreadSafetyMode.NONE) { IonByteArrayImpl(bytes) }
```

Comment:
Is the complexity around making this a lazy field actually worth it?  In other words, I wouldn't bother unless it makes a real difference as the more straightforward code is cleaner.

Reply:
I do think this was a miss on my part.  Fixing in the next revision.